util/normalize.js

```diff
@@ -1,3 +1,20 @@
+const handleVariantGrouping = (prefix) => (acc, x) => {
+  if (prefix) {
+    x = prefix + x;
+    if (x.endsWith(')')) {
+      x = x.slice(0, -1);
+      prefix = null;
+    }
+  } else {
+    const start = x.match(/(.*:)+\(/);
```

Comment:
may want to warn/test against people trying to nest groups:

```
ow`
sm:(
	hover:(
		text-blue-500
	)
)
`
```

Reply:
Good thinking.. I guess this kind of nesting could be supported 🤔 but nesting `sm:` inside of `md:` doesn't make sense, so would/could still warn about that. Will have a stab at this today! Thanks for taking the time to review 🙇

Reply:
np! thanks for picking up on a random idle idea i had. def will shout it out in my post. 

random other thought - did you consider `[ ]` rather than `( )`? `[ ]` says to me that arrays are allowed, whereas `( )` kinda looks weird without commas between them. 🤷🏽  really minor thought

Reply:
Haha, you made a good point about nesting.. once mentioned, an implementation without it wasn't worth merging I didn't think! As you can see [it didn't take much more code to support](https://github.com/lukejacksonn/oceanwind/pull/24/commits/3c1e19d7195546a6a8d362895fc23aa862c4bee7) nesting as you describe above.

Never really give much consideration to either `[ ]` or `( )` really. Just stuck with the first thing I tried which was probably verbatim from your post 😅 . Technically either (or both) could work and both look equally weird without commas now you mention it too, hah.

Thanks for the link from your post. Very much appreciate the exposure as a lot of work has gone into this project so far!